lib/generic/PoleEffect.cpp

```diff
@@ -0,0 +1,38 @@
+#include "PoleEffect.hpp"
+
+#include <FastLED.h>
+
+void PoleEffect::SetPoleEffect(Poles& poles, long t) {
+  for (auto& pole : poles) {
+    pole.ClearPoleLights();
+
+    for (uint8_t i = 0; i < kLightsPerPole; ++i) {
+      CHSV color = GetColor(i, t);
+      pole.SetPoleLight(i, color);
+    }
+  }
+}
+
+void PoleEffect::SetPoleLowerEffect(Poles& poles, long t) {
+  for (auto& pole : poles) {
+    pole.ClearPoleLights();
+
+    for (uint8_t i = 0; i < kLightsPerPole; ++i) {
+      CHSV color = GetColor(i, t);
+      // Light index from [0, 256)
+      uint16_t amount = ((uint16_t)i * 255) / kLightsPerPole;
+      color.v = (color.v * amount) / 255;
+      color.v /= 2;
+      color.v += 128;
```

Comment:
What would you name this operation, collectively? I think it would help me understand what is going on here if this were a function that returns `CHSV` and were named what it is doing. You're getting the color the same as in `SetPoleEffect`, but then doing some extra transformation.

Reply:
I think naming the operation is difficult (make-less-brigher-with-linear-scale) but I can add comments to describe what's going on.